docs/install-service/setup-openshift-390.md

```diff
@@ -0,0 +1,396 @@
+- Create a user that’s able to use the OpenShift web console (on openshift master):
+  ```bash
+  $ oc login -u system:admin --config ~/.kube/config
+  $ oc create user cccp
+  $ htpasswd /etc/origin/master/htpasswd cccp
+  ```
+
+- Create a project:
+  ```bash
+  # create cccp project using cccp user
+  $ oc login -u cccp
+  $ oc new-project cccp
+  ```
+
+- Create a PV using the 5GB NFS share we created earlier. This can be done only
+  as a cluster-admin user. Replace NFS VM IP address to correct one in
+  following yml template:
+  ```bash
+  $ cat jenkins-pv.yml
+  apiVersion: v1
+  kind: PersistentVolume
+  metadata:
+  name: jenkins
+  namespace: cccp
+  spec:
+  capacity:
+  storage: 5Gi
+  accessModes:
+  - ReadWriteOnce
+  nfs:
+  path: /jenkins
+  server: 172.29.33.8
+  persistentVolumeReclaimPolicy: Recycle
+
+  $ oc login -u system:admin --config ~/.kube/config
+
+  $ oc create -f jenkins-pv.yml
+  ```
+- Bring up a Jenkins server using the jenkins persistent template. This has to
+  be done as normal user and not as admin user. ​ Make sure that the
+  Jenkins PV you created in above step is formated/emptied. ​ I spent about
+  2 days trying to debug issues arising out of stale data on that NFS storage.
+  Also add an extra environment variable to make sure that parallel builds work
+  as expected:
+
+  ```bash
+  $ oc login -u cccp
+  $ oc process -p MEMORY_LIMIT=1Gi openshift//jenkins-persistent | oc create -f -
+
+  # add env var for parallel builds
+  $ oc set env dc/jenkins JENKINS_JAVA_OVERRIDES="-Dhudson.slaves.NodeProvisioner.initialDelay=0,-Dhudson.slaves.NodeProvisioner.MARGIN=50,-Dhudson.slaves.NodeProvisioner.MARGIN0=0.85"
+
+  # add required permissions
+  $ oc login -u system:admin
+  $ oc adm policy add-scc-to-user privileged system:serviceaccount:cccp:jenkins
+  $ oc adm policy add-role-to-user system:image-builder system:serviceaccount:cccp:jenkins
+
+  ```
+
+Now go to the OpenShift web console and check if Jenkins deployment has come up
+and if the PVC of that deployment is bound to the PV you created earlier.
+
+To access Jenkins web console, you’ll need to check the OpenShift web console for the
+route that was created with nip.io subdomain. This should be visible on the
+project/namespace’s homepage on OpenShift console.
+
+Configuring DaemonSet: Scanning is one of the build pipeline phase the service
+offers. In scanning, we introspect the image built. In order to make scanning
+module available on all the possible builder nodes, we configure and deploy
+DaemonSet. The DeamonSet spins up a pod per builder node, which avails a docker
+volume for all the containers on the node. The scan stage in pipeline uses the
+volume for performing scan phase. DaemonSet needs to be deployed using cluster
+admin. Configure it with cluster admin user:
+
+```bash
+$ git clone https://github.com/dharmit/ccp-openshift
+$ cd ccp-openshift
+$ oc login -u system:admin
+$ oc create -f daemon-set/scan_data.yaml
```

Comment:
We must add the `oc annotate` command before this to make sure that scan_data daemon set successfully spins up pods on all the nodes. This step is not required for 3.7.0 installation however.

Reply:
Done